2015/proposals/girdhar-yashasvi.md

```diff
@@ -0,0 +1,183 @@
+* Creating a model for storing the data in a relational database.
+
+The first step of the project is deciding what all information is useful to us and how we would be storing it. I have already built a first draft of the schema here, with feedback from the mentor.
+
+* Designing an API to collect the results of the tests.
+
+This step involves designing an API on the server, so that the clients can submit the results through the installation test scripts. The API would collect the results and store them in a 
+relational database, following the schema decided in the first phase.
+I am planning to use Flask + SqlAlchemy for this purpose.
+
+* Modifying the installation test scripts for sending the data to the server.
+
+This step involves sending the required data to the server, in a form that conforms with the API. I have already understood the script and have also submit a PR, that adds documentation explaining the implementation details. 
+
+* API for performing queries on the data (includes building an interface for the target users. )
+
+This step mainly involves three parts : 
+* designing an API to perform queries on the stored data
+* building an interface for the user to use that API, and 
+* adding a data visualisation library for better viewing of the results.
+
+I plan on writing a custom UI in Angularjs that allows the user to construct the queries and then use a data visualisation library for viewing the results.
+
+##Schedule of Deliverables
+
+I plan to complete the back-end before mid term. So that, the latter half will be used for the front end and testing.
+
+##### 27th April to 23rd May : Community bonding period
+
+I would use this period, partially for learning more about the technologies that I would be working on and partially for understanding more about how software carpentry works. I believe that it’s important, so that I believe in its vision and have an inspiration to stay connected with the organisation even after the summer of code ends. I do not want that my work should be a matter of just 3 months.
+
+I have been learning the technologies that I mentioned since past 4-5 days, and will continue that till the project starts. I think I have enough time to get a good hold of them before the project starts.
+
+Moving on, this period is important for us to decide what all information can be useful for us : 
+I have already discussed that, will update if there happens to be any change in the requirements 
+
+##### 1) API for pushing the data to the server (1 week - 10 days) : 23th May to 1st june
+
+* decide what all we need to send
+* decide how to send. most probably using urllib2 so that no extra dependency is introduced
+* decide in which form the data should be sent
+
+##### 2) Build the schema: (3-4 days) : 1st June to 5 June
+
+I have already built a draft of the schema [here](https://docs.google.com/document/d/1RYKx_V7uy8Z8W33VKpIMaXwO8GnkoCE4KDDegD6gIqk/edit) with some feedback, so it shouldn’t take much time.
```

Comment:
I'd inline the schema, but maybe in a different section of the proposal to avoid cluttering the schedule.  That's one less link to click for readers, and it's easier for me to review it in GitHub than via Google Docs.